src/node_report_utils.cc

```diff
@@ -0,0 +1,491 @@
+  } else {
+    // Parse the supplied switch
+    if (!strncmp(args, "SIGUSR2", sizeof("SIGUSR2") - 1)) {
+      return SIGUSR2;
+    } else if (!strncmp(args, "SIGQUIT", sizeof("SIGQUIT") - 1)) {
+      return SIGQUIT;
+    } else {
+     std::cerr << "Unrecognised argument for node-report signal option:"
+               << args
+               << "\n";
+    }
+  }
+  return SIGUSR2;  // Default signal is SIGUSR2
+#endif
+}
+
+/*******************************************************************************
+ * Function to process node-report config: specification of report file name.
+ ******************************************************************************/
+void ProcessNodeReportFileName(const char* args) {
+  if (strlen(args) == 0) {
+    std::cerr << "Missing argument for node-report filename option\n";
+    return;
+  }
+  if (strlen(args) > NR_MAXNAME) {
+    std::cerr << "Supplied node-report filename too long (max "
+              << NR_MAXNAME
+              << " characters)\n";
+    return;
+  }
+  snprintf(report_filename, sizeof(report_filename), "%s", args);
+}
+
+/*******************************************************************************
+ * Function to process node-report config: specification of report directory.
+ ******************************************************************************/
+void ProcessNodeReportDirectory(const char* args) {
+  if (strlen(args) == 0) {
+    std::cerr << "Missing argument for node-report directory option\n";
+    return;
+  }
+  if (strlen(args) > NR_MAXPATH) {
+    std::cerr << "Supplied node-report directory path too long (max "
+              << NR_MAXPATH
+              << " characters)\n";
+    return;
+  }
+  snprintf(report_directory, sizeof(report_directory), "%s", args);
+}
+
+/*******************************************************************************
+ * Function to process node-report config: verbose mode switch.
+ ******************************************************************************/
+unsigned int ProcessNodeReportVerboseSwitch(const char* args) {
+  if (strlen(args) == 0) {
+    std::cerr << "Missing argument for node-report verbose switch option\n";
+    return 0;
+  }
+  // Parse the supplied switch
+  if (!strncmp(args, "yes", sizeof("yes") - 1) ||
+      !strncmp(args, "true", sizeof("true") - 1)) {
+    return 1;
+  } else if (!strncmp(args, "no", sizeof("no") - 1) ||
+             !strncmp(args, "false", sizeof("false") - 1)) {
+    return 0;
+  } else {
+    std::cerr << "Unrecognised argument for node-report verbose switch option: "
+              << args
+              << "\n";
+  }
+  return 0;  // Default is verbose mode off
+}
+
+/*******************************************************************************
+ * Function to obtain the node and subcomponent version strings.
+ *******************************************************************************/
+std::string GetComponentVersions() {
```

Comment:
Since this is no longer called when node-report initializes and instead is called every time the report is written, perhaps just move this to `src/node_report.cc` and write directly to the stream instead of creating an intermediary string.

Reply:
thanks, done.